File: core/src/com/joshwiegand/portal/asteroids/Application.java
package com.joshwiegand.portal.asteroids;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.joshwiegand.portal.asteroids.game.GameControl;
import com.joshwiegand.portal.asteroids.game.GameModel;
import com.joshwiegand.portal.asteroids.util.Assets;

public class Application extends ApplicationAdapter {
    private SpriteBatch batch;
    private com.joshwiegand.portal.asteroids.game.GameModel model;

    @Override
    public void create() {
        Assets.init();
        batch = new SpriteBatch();
        GameControl controller = new GameControl();
        model = new GameModel(controller);
        Gdx.input.setInputProcessor(controller);
    }

    @Override
    public void dispose() {

    }

    @Override
    public void pause() {

    }

    @Override
    public void render() {
        model.update();
        Gdx.gl.glClearColor(0, 0, 0, 0);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);

        batch.begin();
        batch.disableBlending();
        batch.draw(Assets.BG, 0, 0);
        batch.enableBlending();

        for (Sprite s : model.getSprites())
            s.draw(batch);

        String displayPanel = "Score: " + model.getScore() + "   Lives: " + model.getLives();
        Assets.FONT.draw(batch, displayPanel, 0, Gdx.graphics.getHeight());
        batch.end();
    }

    public static void explosion() {
        Assets.PLAYER_EXPLOSION.play();
    }

    public static void shot() {
        Assets.SHOT.play();
    }

    @Override
    public void resize(int arg0, int arg1) {

    }

    @Override
    public void resume() {

    }
}

File: core/src/com/joshwiegand/pa/Application.kt
package com.joshwiegand.pa

import com.badlogic.gdx.ApplicationAdapter
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.joshwiegand.pa.game.GameControl
import com.joshwiegand.pa.game.GameModel
import com.joshwiegand.pa.util.Assets

class Application : ApplicationAdapter() {
    private var batch: SpriteBatch? = null
    private var model: com.joshwiegand.pa.game.GameModel? = null

    override fun create() {
        Assets.init()
        batch = SpriteBatch()
        val controller = GameControl()
        model = GameModel(controller)
        Gdx.input.inputProcessor = controller
    }

    override fun dispose() {

    }

    override fun pause() {

    }

    override fun render() {
        model!!.update()
        Gdx.gl.glClearColor(0f, 0f, 0f, 0f)
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)

        batch!!.begin()
        batch!!.disableBlending()
        batch!!.draw(Assets.BG, 0f, 0f)
        batch!!.enableBlending()

        for (s in model!!.sprites)
            s.draw(batch!!)

        val displayPanel = "Score: " + model!!.score + "   Lives: " + model!!.lives
        Assets.FONT.draw(batch, displayPanel, 0f, Gdx.graphics.height.toFloat())
        batch!!.end()
    }

    override fun resize(arg0: Int, arg1: Int) {

    }

    override fun resume() {

    }
}

